lib/core/src/Cardano/Wallet/Primitive/AddressDerivation/Random.hs

```diff
@@ -369,3 +442,31 @@ decodeNestedBytes dec bytes =
         Right ("", res) -> pure res
         Right _ -> fail "Leftovers when decoding nested bytes"
         _ -> fail "Could not decode nested bytes"
+
+addrToPayload :: Address -> ByteString
+addrToPayload (Address addr) =
+    unsafeDeserialiseFromBytes decodeAddressPayload $ b58decode addr
+
+-- | CBOR deserialise without error handling - handy for prototypes or testing.
+unsafeDeserialiseFromBytes :: (forall s. CBOR.Decoder s a) -> BL.ByteString -> a
+unsafeDeserialiseFromBytes decoder bytes =
+    either (\e -> error $ "unsafeDeserialiseFromBytes: " <> show e) snd $
+        CBOR.deserialiseFromBytes decoder bytes
+
+-- | Decode a bitcoin base-58 address to a LBS, without error handling.
+b58decode :: ByteString -> BL.ByteString
+b58decode = BL.fromStrict . fromJust . decodeBase58 bitcoinAlphabet
```

Comment:
`Base58` encoding is actually something that is specific to the http-bridge / byron network. In _theory_, it'd be totally valid to encode random addresses using bech32. The address derivation scheme and its user-facing encoding sits at two different levels in the abstraction and shouldn't be conflated.